ddp-workspace/projects/ddp-dsm-ui/src/app/scanner/scanner.component.ts

```diff
@@ -169,14 +171,29 @@ export class ScannerComponent implements DoCheck, OnDestroy {
   }
 
   private removeSuccessfulScans(responseData: any[]): void {
-    responseData.forEach((data: any, index: number) => data &&
-      this.scannerFields.at(index).setErrors({notFound: data?.error})
+    responseData.forEach((data: any, index: number) => {
+    if (data?.hasOwnProperty('error')) {
+        this.scannerFields.at(index).setErrors({notFound: data?.error});
+        this.additionalMessage = 'Error - Failed to save all changes';
+        }
+      }
+    );
+
+    responseData.forEach((data: any, index: number) =>
+      {
+      data?.hasOwnProperty('shortId') && this.setShortId(index, data);
+      }
     );
+
     for (let i = this.scannerFields.length - 2; i >= 0; i--) {
-      !this.scannerFields.at(i).hasError('notFound') && this.scannerFields.removeAt(i);
+      !this.scannerFields.at(i).hasError('notFound') && !this.shortIds[i] && this.scannerFields.removeAt(i);
```

Comment:
What do you think about setting the different type of error, instead of the `notFound`, in which case you would not need to create a separate array and render it separately in the UI? 
You would only need to check for that error and act accordingly

Reply:
Since this is not an error, I don't think we should add it to `hasError`